### PR TITLE
feat(python): support python 3.13

### PR DESCRIPTION
**Pull request by cpcloud**
Add support Python 3.13

### CODE REVIEW

File: .github/workflows/nix.yml [hunk 1]
```diff
@@ -38,11 +38,14 @@ jobs:
...
           - "3.11"
           - "3.12"
+          - "3.13"
         include:
           - os: ubuntu-arm64-24.04 # aarch64-linux
             python-version: "3.12"
           - os: macos-14 # aarch64-darwin
             python-version: "3.12"
+          - os: macos-14 # aarch64-darwin
+            python-version: "3.13"
```
Comment (gforsyth) [inline .github/workflows/nix.yml:48]: Does this need a corresponding entry in the skip helper?

**Review by gforsyth — APPROVED** [review]
nice!

File: .github/workflows/nix.yml (hunk 1 above)
Comment (cpcloud) [reply .github/workflows/nix.yml:48]: Done.

File: nix/overlay.nix [hunk 2]
```diff
@@ -53,6 +53,11 @@ let
         # Use base package set from pyproject.nix builders
         (pkgs.callPackage pyproject-nix.build.packages {
           inherit python;
+          stdenv = pkgs.stdenv.override {
+            targetPlatform = pkgs.stdenv.targetPlatform // {
+              darwinSdkVersion = "12.0";
```
Comment (cpcloud) [inline nix/overlay.nix:58]: This isn't strictly required for python 3.13 support, but it sure does make life easier by allowing more wheels to be used on macOS. A bunch of the more-complex-to-build wheels like scipy and scikit-learn are only available using a 12.0 darwin SDK build.

**Comment by cpcloud** [comment]
The remaining single failure was fixed in https://github.com/kylebarron/arro3/issues/276 and once the 0.4.5 wheels are uploaded (after [this CI run](https://github.com/kylebarron/arro3/actions/runs/12356587820) succeeds), then I'll update `uv.lock` using `just update` and we should be good to merge this PR.

**Comment by cpcloud** [comment]
Ok, bumped those dumps by way of `just update`.

**Comment by cpcloud** [comment]
I'll do some follow up to see whether we can run any of the clouds with 3.13.

**Pull request merged**